libs/utils/src/JobSystem.cpp

```diff
@@ -105,30 +105,34 @@ void JobSystem::setThreadAffinityById(size_t id) noexcept {
 #endif
 }
 
-JobSystem::JobSystem(size_t threadCount, size_t adoptableThreadsCount) noexcept
+JobSystem::JobSystem(const size_t userThreadCount, const size_t adoptableThreadsCount) noexcept
     : mJobPool("JobSystem Job pool", MAX_JOB_COUNT * sizeof(Job)),
       mJobStorageBase(static_cast<Job *>(mJobPool.getAllocator().getCurrent()))
 {
     SYSTRACE_ENABLE();
 
-    if (threadCount == 0) {
+    int threadPoolCount = userThreadCount;
+    if (threadPoolCount == 0) {
         // default value, system dependant
-        size_t hwThreads = std::thread::hardware_concurrency();
+        int hwThreads = std::thread::hardware_concurrency();
         if (UTILS_HAS_HYPER_THREADING) {
             // For now we avoid using HT, this simplifies profiling.
             // TODO: figure-out what to do with Hyper-threading
-            threadCount = hwThreads / 2 - 1;
-        } else {
-            threadCount = hwThreads - 1;
+            // since we assumed HT, always round-up to an even number of cores (to play it safe)
+            hwThreads = (hwThreads + 1) / 2;
         }
+        // make sure we have at least one h/w thread (could be an assert instead)
+        hwThreads = std::max(0, hwThreads);
```

Comment:
When can hwThread ever be negative? Did you mean max(1,...)? That's what the comment says.

Reply:
it should never happen, that's why "it could be an assert".